src/tini.c

```diff
@@ -470,6 +476,12 @@ int reap_zombies(const pid_t child_pid, int* const child_exitcode_ptr) {
 						PRINT_FATAL("Main child exited for unknown reason");
 						return 1;
 					}
+				} else {
+					if (warn_on_reap > 0) {
+						PRINT_WARNING("Reaped child with pid: '%i'", current_pid);
+					} else {
+						PRINT_DEBUG("Reaped child with pid: '%i'", current_pid);
```

Comment:
I'd rather keep the debug above so that it logs unconditionally (including for the "main child"), but have a `PRINT_WARNING` here dedicated to the case where the child is *not* the main child (i.e. this branch you're in).

We can also make the the warning more explicit. E.g. "Reaped orphaned process with pid: '%i'".